corehq/apps/users/views/utils.py

```diff
@@ -113,26 +114,46 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
         )
 
 
-def get_locations_with_cases(domain, location_ids):
-    """
-    Takes a list of location IDs and returns how many cases are assigned to each location
-    :param domain: The domain to search in
-    :param location_ids: A list of location IDs to get case counts on
-    :returns A dict with location names as the key, and the number of cases assigned to it as the value
-    """
+def _get_location_ids_with_single_user(domain, location_ids, user_id):
+    other_commcare_user_ids = [user.user_id for user in get_all_commcare_users_by_domain(domain)]
+    # Remove the user's ID, as we want to see which other CommCare users share any of the given location_ids
+    other_commcare_user_ids.remove(user_id)
     query = (CaseES()
             .domain(domain)
             .owner(location_ids)
+            .user(other_commcare_user_ids)
             .aggregation(TermsAggregation('by_location', 'owner_id')
-            .size(0)))
-    counts = query.run().aggregations.by_location.counts_by_bucket()
-    locations_with_cases = dict()
-    if counts:
-        locations = SQLLocation.objects.filter(location_id__in=location_ids)
+            .size(0))
+    ).run()
+    return list(set(location_ids) - set(query.aggregations.by_location.keys))
+
+
+def get_locations_with_single_user(domain, location_ids, user_id):
+    """
```

Comment:
Will be good to have a test for this.

Reply:
Unit tests have been created for the helper function.